nautobot/ui/src/components/common/BSNavBar.jsx

```diff
@@ -1,17 +1,33 @@
-import { Container, Nav, Navbar, NavDropdown } from "react-bootstrap"
-import { faRightToBracket } from "@fortawesome/free-solid-svg-icons";
+import { Button, Container, Nav, Navbar, NavDropdown } from "react-bootstrap"
+import { faSignIn, faSignOut } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { Link, NavLink } from "react-router-dom"
+import { Link, NavLink, redirect, useNavigate } from "react-router-dom"
+import {useState, useEffect} from "react"
 import useSWR from "swr"
+import axios from "axios";
 
 
 const fetcher = (url) => fetch(url, { credentials: "include" }).then((res) => res.json());
 
 export default function BSNavBar() {
+  const navigate = useNavigate();
+
   const { data, error } = useSWR("/api/get-menu/", fetcher)
+  const [ isLoggedIn, setIsLoggedIn] = useState(false)
+  useEffect(() => {
+    setIsLoggedIn(localStorage.getItem("nautobot-user") != null)
+  }, [])
```

Comment:
Can you add some comments for this block of code pretty pls? Just wondering what it does.